File: project2/Hogweed.java
class Hogweed extends Plant {
    private static final int WIDTH = 40;
    private static final int HEIGHT = 20;
    
    public Hogweed(int x, int y, World w) {
        super(10, x, y, w);
    }
    
    @Override
    public char draw() {
        return 'h';
    }
    
    @Override
    public void action() {
        for (int dx = -1; dx <= 1; dx++) {
            for (int dy = -1; dy <= 1; dy++) {
                int nx = x + dx;
                int ny = y + dy;
                if (nx >= 0 && ny >= 0 && nx < WIDTH && ny < HEIGHT) {
                    Organism target = world.grid[ny][nx];
                    if (target != null && target instanceof Animal) {
                        target.alive = false;
                        world.grid[ny][nx] = null;
                        world.lastCollisionMessage = draw() + " killed " + target.draw() +
                                                     " at (" + nx + ", " + ny + ")";
                    }
                }
            }
        }
        super.action();
    }
}

File: project3/hogweed.py
from plant import Plant
from animal import Animal

class Hogweed(Plant):
    def __init__(self, x, y, world):
        super().__init__(10, x, y, world)
    
    def action(self):
        # Kill all animals in adjacent cells
        directions = [(0,0), (0,1), (1,0), (0,-1), (-1,0)]
        
        for dx, dy in directions:
            nx, ny = self.x + dx, self.y + dy
            if 0 <= nx < self.WIDTH and 0 <= ny < self.HEIGHT:
                organism = self.world.grid[ny][nx]
                if organism is not None and isinstance(organism, Animal) and organism.draw() != 'C':
                    organism.alive = False
                    self.world.grid[ny][nx] = None
                    self.world.last_collision_message = f"Hogweed killed {organism.draw()} at ({nx}, {ny})"
        
        super().action()
    
    def draw(self):
        return 'h'
